AG-16452 - High-frequency Example Improvements

alantreadway opened the pull request:
https://ag-grid.atlassian.net/browse/AG-16452

- QoL: For dev-mode add commented production and staging URLs to allow quick switching in `vanilla` examples.
- Improve state management for high-freq-* examples (reduce boilerplate).
- Introduce in-example benchmark harness for `high-freq-high-volume` as a prototype for browser-based benchmarking.

alantreadway committed:
refactor: unify form control persistence in high-frequency examples

Replace per-control persistence functions with unified config object pattern:
- Single FormConfig interface for all form state
- Unified loadConfig() and saveConfig() functions
- Hash-only persistence (removed sessionStorage)
- Proper initialization order: config loads before intervals/RAF start
- Only non-default values persisted to hash

Affected examples:
- high-freq-high-volume
- high-freq-random-update
- high-freq-financial-random-update
- high-freq-financial-high-volume

Reduces boilerplate by ~150-200 lines per example and improves maintainability.

alantreadway committed:
Add commented UMD script tags to vanilla dev examples

Inject commented staging and production UMD URLs into generated vanilla example index.html files during dev builds. URLs are selected based on whether the example uses community or enterprise packages.

alantreadway committed:
Extract benchmark module and improve UI

- Extract benchmark functionality into reusable benchmark.ts module
- Add configuration metadata capture to benchmark JSON export
- Condense benchmark UI to single horizontal line with badges
- Remove redundant inner box styling
- Add status badge (grey while running, green when complete)
- Float version badge to the right
- Remove 'Benchmark Running' heading

alantreadway committed:
nx format

alantreadway committed:
Fix TypeScript errors: prefix unused parameters with underscore

lsjroberts commented on plugins/ag-charts-generate-example-files/src/executors/generate/generator/utils/frameworkFilesGenerator.ts:
```diff
@@ -11,6 +11,22 @@ import type { FileContents, InternalFramework } from '../types';
+        ? 'https://charts-staging.ag-grid.com/dev/ag-charts-enterprise/dist/umd/ag-charts-enterprise.js'
+        : 'https://charts-staging.ag-grid.com/dev/ag-charts-community/dist/umd/ag-charts-community.js';
+    const productionUrl = isEnterprise
+        ? 'https://cdn.jsdelivr.net/npm/ag-charts-enterprise@13.0.0/dist/umd/ag-charts-enterprise.js'
```
`ag-charts-enterprise@latest`?

alantreadway replied:
I was thinking that it's quicker to tweak the version number here in this format - but I could make it dynamic at the least

alantreadway replied:
Updated to dynamically insert the latest version

lsjroberts commented on tsconfig.base.json:
were you trying to ignore prettier on this file? undo the changes?

alantreadway replied:
This seems to be a continuing source of false formatting, the change here is just switching to the format that Nx forces, so future `nx format` runs should then leave this file unaffected 🤞

lsjroberts reviewed (approved):
LGTM

alantreadway committed:
Enhance benchmark functionality with cancellation support and UI improvements

- Added `onComplete` callback to restore original series type after benchmark completion.
- Implemented a `stop` method to allow users to cancel ongoing benchmarks.
- Updated UI to reflect benchmark progress and include a stop button.
- Improved event delegation for stop button clicks to enhance user experience.

alantreadway committed:
Make UMD script tag version dynamic based on package.json

- Read version from package.json at build time
- Strip any -beta.XXXXXXX suffix from version
- Use dynamic version for both enterprise and community production URLs
